config-v2-documentation

```diff
@@ -88,6 +90,19 @@ tokens_float_abs_error=1e-30
 # If used, both of these options must be specified
 # To explicitly disable float checking, set both options to the empty string
 
+shuffle_mode=not_set
+# Only for out_check=shuffle
+# Which permutations are allowed:
+#   lines       Permutation of lines
+#   words       Permutation of words within each line
+#   lines_words Both lines and words
+#   tokens      Permutation of all tokens, ignoring line boundaries
+
+shuffle_ignore_case=0
```

Comment:
If we would have `ignore_case` for all `out_check`s, then it would make sense. But in this case it makes the config more clear having the two keys separate.